src/core/FxMixer.cpp

```diff
@@ -327,6 +327,19 @@ void FxMixer::deleteChannel( int index )
 	for( int i = index; i < m_fxChannels.size(); ++i )
 	{
 		validateChannelName( i, i + 1 );
+
+		// set correct channel index
+		m_fxChannels[i]->m_channelIndex = i;
```

Comment:
This looks stupid, and indeed it is.

I did not add any behaviour here, though. This is just inlining what was the previous `validateChannelName` function.

I did not analyze whether this could be removed or not, mantained consistency with the previous code.